### PR TITLE
feat(rich-text-editor): DLT-2244 add ability inherit image properties in editor

### PR DESCRIPTION
**Pull request by emariotti3**
# feat(rich-text-editor): DLT-2244 add ability inherit image properties in editor

## Obligatory GIF (super important!)

Obligatory gif

## :hammer_and_wrench: Type Of Change

These types will increment the version number on release:

- [ ] Fix
- [x] Feature
- [ ] Performance Improvement
- [ ] Refactor

## :book: Jira Ticket

[DLT-2244](https://dialpad.atlassian.net/browse/DLT-2244)

## :book: Description

Tip tap seems to strip the width from inserted images, so we need to add support adjusting image sizing. We found this useful thread [here](https://github.com/ueberdosis/tiptap/issues/4103) showing how to get the tiptap image extension to inherit the attributes from the img html tag.

## :bulb: Context

This is mainly to be able to use it in email reply quotes and perhaps we could use it as well to add the ability to insert inline attachments/images. See short video [here](https://drive.google.com/file/d/1a_lfH2hMUusQeGCwZaYKBG4Pq3bbbRiM/view?usp=sharing) showing the problem.

## :pencil: Checklist

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.
- [x] I have added all relevant documentation.
- [x] I have considered the performance impact of my change.

For all Vue changes:

- [x] I have added / updated unit tests.
- [x] I have made my changes in Vue 2 and Vue 3. Note: you may sync your changes from Vue 2 to Vue 3 (or vice versa) using the `./scripts/dialtone-vue-sync.sh` script. Read docs here: [Dialtone Vue Sync Script](../packages/dialtone-vue3/.github/CONTRIBUTING.md#dialtone-vue-sync-script)
- [x] I have validated components with a screen reader.
- [x] I have validated components keyboard navigation.

## :camera: Screenshots / GIFs

DLT-2244-add-ability-resize-images-editor

DLT-2244-add-ability-resize-images-editor-2

## :link: Sources

Adding the link for the tip-tap [issue](https://github.com/ueberdosis/tiptap/issues/4103) referenced for this change

[DLT-2244]: https://dialpad.atlassian.net/browse/DLT-2244?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/dialtone-vue3/components/rich_text_editor/rich_text_editor_default.story.vue [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <template>
   <dt-rich-text-editor
-    v-model="value"
+    v-model="$attrs.value"
```
Comment (juliodialpad) [inline packages/dialtone-vue3/components/rich_text_editor/rich_text_editor_default.story.vue]: This is not correct, the value is defined in the component data on line 43, so it should only be

```suggestion
    v-model="value"
```

we only add the `$attrs` when this is coming from storybook

File: packages/dialtone-vue3/recipes/conversation_view/editor/editor.vue [hunk 2]
```diff
@@ -151,7 +151,7 @@
 
     <!-- Some wrapper to restrict the height and show the scrollbar -->
     <div
-      class="d-of-auto d-mx12 d-mt12 d-mb16 d-c-text"
+      class="d-of-auto d-mx16 d-mb4 d-c-text"
```
Comment (juliodialpad) [inline packages/dialtone-vue3/recipes/conversation_view/editor/editor.vue]: Thanks for this fix, haven't noticed this was out of sync between vue 2 and vue 3 versions

File: packages/dialtone-vue3/recipes/conversation_view/editor/editor_default.story.vue [hunk 3]
```diff
@@ -2,7 +2,7 @@
   <div>
     <dt-recipe-editor
       ref="editor"
-      v-model="modelValue"
+      v-model="$attrs.value"
```
Comment (juliodialpad) [inline packages/dialtone-vue3/recipes/conversation_view/editor/editor_default.story.vue]: remove `$attrs`

```suggestion
      v-model="value"
```

File: packages/dialtone-vue3/recipes/conversation_view/editor/editor_default.story.vue [hunk 4]
```diff
@@ -46,7 +46,7 @@ export default {
   components: { DtRecipeEditor },
   data () {
     return {
-      modelValue: this.$attrs.modelValue,
+      value: this.$attrs.value,
```
Comment (juliodialpad) [inline packages/dialtone-vue3/recipes/conversation_view/editor/editor_default.story.vue:49]:
```suggestion
      value: this.$attrs.modelValue,
```

**Comment by juliodialpad** [comment]
Hi @emariotti-dialpad Please let me know how to properly test this, I tried just copy pasting an image into the editor and it's just pasting a giant icon. 

Screenshot 2024-12-27 at 12 33 11 p m

**Review by juliodialpad — CHANGES_REQUESTED** [review]
The value is not being updated properly on Vue 3, please check that Vue 2 and Vue 3 versions are working properly on storybook.

**Comment by emariotti3** [comment]
@juliodialpad I run it locally and set the image tag in the `value` string in the argsData parameter here `packages/dialtone-vue3/recipes/conversation_view/editor/editor.stories.js`

Screenshot 2024-12-27 at 5 24 04 PM

**Review by juliodialpad — APPROVED** [review]
Thanks! working correctly now

**Pull request merged**